src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -43,13 +43,23 @@ const Body = BaseForm.extend(Object.assign(
       this.deviceChallengePollRemediation = this.options.currentViewState;
       this.doChallenge();
       this.startPolling();
+
+      this.listenTo(this.options.appState, 'change:dynamicRefresh', this.updateRefresh);
     },
 
     onPollingFail () {
       this.$('.spinner').hide();
       this.stopPolling();
     },
 
+    updateRefresh () {
+      if (this.polling) {
+        this.options.currentViewState = this.options.appState.get('dynamicRefresh');
```

Comment:
Changing the `currentViewState` this way a bit _hacky_.
I was more leaning toward to only pass around the new refresh interval; then change the `startPolling` function
like
```
this.startPolling(newRefresh);

-----
  startPolling (newInterval) {
    this.pollingInterval = newInterval  || this.options.currentViewState.refresh;
  ...
}
```

what do you think @pradeepdewda-okta @jane-kuang-okta

Reply:
FYI, will be keeping as is in latest commit. ^Comment above was based on outdated code.